sklearn/tests/test_common.py

```diff
@@ -297,3 +304,192 @@ def test_search_cv(estimator, check, request):
+            setattr(self, key, value)
+        return self
+
+    def __getstate__(self):
+        return self.__dict__.copy()
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    def fit(self, X, y):
+        X, y = check_X_y(X, y)
+        self.n_features_in_ = X.shape[1]
+        self._mean = np.mean(y)
+        return self
+
+    def predict(self, X):
+        check_is_fitted(self)
+        X = check_array(X)
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError
+        return np.ones(shape=(X.shape[0],)) * self._mean
+
+    def score(self, X, y):
+        from sklearn.metrics import r2_score
+        return r2_score(y, self.predict(X))
+
+
+class MinimalTransformer:
+    """Minimal transformer implementation with inheriting from
+    BaseEstimator."""
+
+    def __init__(self, param=None):
+        self.param = param
+
+    def __repr__(self):
+        # Only required when using pytest-xdist to get an id not associated
+        # with the memory location
+        return self.__class__.__name__
+
+    def get_params(self, deep=True):
+        return {"param": self.param}
+
+    def set_params(self, **params):
+        valid_params = self.get_params()
+        for key, value in params.items():
+            if key not in valid_params:
+                raise ValueError("Wrong params")
+            setattr(self, key, value)
+        return self
+
+    def __getstate__(self):
+        return self.__dict__.copy()
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    def fit(self, X, y=None):
+        X = check_array(X)
+        self.n_features_in_ = X.shape[1]
+        return self
+
+    def transform(self, X, y=None):
+        check_is_fitted(self)
+        X = check_array(X)
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError
+        return X
+
+    def inverse_transform(self, X, y=None):
+        return self.transform(X)
+
+    def fit_transform(self, X, y=None):
+        return self.fit(X, y).transform(X, y)
+
+
+def _generate_minimal_compatible_instances():
+    """Generate instance containing estimators from minimal class compatible
+    implementation."""
+    for SearchCV, (Estimator, param_grid) in zip(
+        [GridSearchCV, RandomizedSearchCV],
+        [
+            (MinimalRegressor, {"param": [1, 10]}),
+            (MinimalClassifier, {"param": [1, 10]}),
+        ],
+    ):
+        yield SearchCV(Estimator(), param_grid)
+
+    for SearchCV, (Estimator, param_grid) in zip(
+        [GridSearchCV, RandomizedSearchCV],
+        [
+            (MinimalRegressor, {"minimalregressor__param": [1, 10]}),
+            (MinimalClassifier, {"minimalclassifier__param": [1, 10]}),
+        ],
+    ):
+        yield SearchCV(
+            make_pipeline(MinimalTransformer(), Estimator()), param_grid
+        ).set_params(error_score="raise")
+
+
+# FIXME: hopefully in 0.25
+@pytest.mark.skip(
```

Comment:
I will switch to `xfail` which I think is better At least the coverage should work then.